addons/account/models/account_move_line.py

```diff
@@ -700,9 +700,10 @@ def _compute_same_currency(self):
     @api.depends('currency_rate', 'balance')
     def _compute_amount_currency(self):
         for line in self:
-            if line.amount_currency is False:
+            if line.amount_currency is False or line._origin.currency_id == line.company_id.currency_id:
                 line.amount_currency = line.currency_id.round(line.balance * line.currency_rate)
-            if line.currency_id == line.company_id.currency_id:
+
+            elif line.currency_id == line.company_id.currency_id:
                 line.amount_currency = line.balance
```

Comment:
```py
            if line.currency_id != line.company_id.currency_id:
                line.amount_currency = line.currency_id.round(line.balance * line.currency_rate)
```
...should be enough. Except if we change the logic in 16.0 ( @william-andre cc), we never recompute `amount_currency` from the `balance` but only in the other way

Reply:
Ok don't know if @william-andre has an opinion on it

Reply:
ça casse tout